src/main/java/org/betonquest/betonquest/events/GiveEvent.java

```diff
@@ -53,30 +55,54 @@ protected Void execute(final Profile profile) {
-                } else {
-                    stackSize = amountInt;
-                }
+                final int stackSize = Math.min(amountInt, 64);
                 final ItemStack item = questItem.generate(stackSize, profile);
-                final HashMap<Integer, ItemStack> left = player.getInventory().addItem(item);
-                for (final ItemStack itemStack : left.values()) {
-                    if (Utils.isQuestItem(itemStack)) {
-                        BetonQuest.getInstance().getPlayerData(profile).addItem(itemStack, stackSize);
-                    } else {
-                        player.getWorld().dropItem(player.getLocation(), itemStack);
+                if (backpack) {
+                    if (!addToBackpack(profile, item)) {
+                        final HashMap<Integer, ItemStack> left = player.getInventory().addItem(item);
+                        for (final ItemStack itemStack : left.values()) {
+                            player.getWorld().dropItem(player.getLocation(), itemStack);
+                            notifyPlayer(profile, NotifyType.DROP);
+                        }
                     }
-                    final String type = Utils.isQuestItem(itemStack) ? "backpack" : "drop";
-                    try {
-                        Config.sendNotify(null, profile.getOnlineProfile().get(), "inventory_full_" + type, null, "inventory_full_" + type + ",inventory_full,error");
-                    } catch (final QuestRuntimeException e) {
-                        LOG.warn("The notify system was unable to play a sound for the 'inventory_full_" + type + "' category. Error was: '" + e.getMessage() + "'", e);
+                } else {
+                    final HashMap<Integer, ItemStack> left = player.getInventory().addItem(item);
+                    for (final ItemStack itemStack : left.values()) {
+                        if (addToBackpack(profile, itemStack)) {
+                            notifyPlayer(profile, NotifyType.BACKPACK);
+                        } else {
+                            player.getWorld().dropItem(player.getLocation(), itemStack);
+                            notifyPlayer(profile, NotifyType.DROP);
+                        }
                     }
                 }
-                amountInt = amountInt - stackSize;
+                amountInt -= stackSize;
             }
         }
         return null;
     }
+
+    private boolean addToBackpack(final Profile profile, final ItemStack itemStack) {
+        if (Utils.isQuestItem(itemStack)) {
+            BetonQuest.getInstance().getPlayerData(profile).addItem(itemStack, itemStack.getAmount());
+            return true;
+        }
+        return false;
+    }
+
+    private void notifyPlayer(final Profile profile, final NotifyType type) {
+        try {
+            Config.sendNotify(null, profile.getOnlineProfile().get(), "inventory_full_" + type.getString(), null, "inventory_full_" + type.getString() + ",inventory_full,error");
```

Comment:
I think type.getString() need to be lowercase

Reply:
That should be done with the overriden getString() method.